Navbar rebasement

nsandoya opened the pull request:
This PR introduces some improvements to the `Navbar.astro` component, aligning it with the latest mobile-first design:

- **Refactor of Navbar.astro:** Extracted the mobile menu logic and HTML into a new `OffcanvasMenu.astro` component.
- **Offcanvas menu interactivity:** Rebuilt the original logic using basic Alpine.js directives and custom methods. Also added animations to enhance the user experience.
- **For new contributors:** Included detailed comments in `Navbar.astro`, `OffcanvasMenu.astro`, and `Link.astro` to explain  how these components work.

![navbar_design_reference](https://github.com/user-attachments/assets/46ace991-4699-448b-bb57-70905436d95c)

Resolves #33

bjgavilanes commented on src/components/Navbar/OffcanvasMenu.astro:
When running `biome check` -- more specifically, `pnpm check`, this is the error prompt I got:

```sh
  ✖ Shouldn't redeclare 'Link'. Consider to delete it or rename it.
  
    3 │ import Link from "./Link.astro";
    4 │ 
  > 5 │ interface Link {
      │           ^^^^
    6 │ 	text: string;
    7 │ 	href: string;
  
  ℹ 'Link' is defined here:
  
    1 │ ---
    2 │ import closeIcon from "../../../public/icons/closeIcon.svg?raw";
  > 3 │ import Link from "./Link.astro";
      │     ^^^^
    4 │ 
    5 │ interface Link {
  

Checked 36 files in 15ms. No fixes applied.
Found 1 error.
check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.
```

This is related to this rule:

https://biomejs.dev/linter/rules/no-redeclare/

This an easy fix, actually. Also, the website does compile; but I would prefer this got fixed.

bjgavilanes commented on src/styles/globals.css:
As we are using biome as linter and formatter, it just support vanilla CSS; so we need to put an ";" at the end of the line.

Related:

https://github.com/biomejs/biome/discussions/4141

bjgavilanes replied:
https://github.com/biomejs/biome/discussions/4141 this one, too.

bjgavilanes reviewed (changes requested):
I got two comments in the codebase; I fixed one locally -- the one at global.css -- and I'm going to push to this PR branch.

Also, while to other issue is not-that-critical, I would prefer it to be fixed.

Again, the website does compile. Thank you for everything!

nsandoya commented:
> I got two comments in the codebase; I fixed one locally -- the one at global.css -- and I'm going to push to this PR branch.
> 
> Also, while to other issue is not-that-critical, I would prefer it to be fixed.
> 
> Again, the website does compile. Thank you for everything!

Hi @bjgavilanes,
Thanks for your detailed review! 🌱
If you'd like, I can implement the changes you suggested in the CSS file and the Link.astro interface to save you some time

bjgavilanes commented:
> Thanks for your detailed review! 🌱
> If you'd like, I can implement the changes you suggested in the CSS file and the Link.astro interface to save you some time

Actually, I already implement the changes in CSS file. It should be in the git log; at least, in mine is. Let me check.

Can you go ahead with Link.astro, please?

Thanks!,
Brayen

nsandoya commented:
> Thanks for your detailed review! 🌱 If you'd like, I can implement the changes you suggested in the CSS file and the Link.astro interface to save you some time
> Actually, I already implement the changes in CSS file. It should be in the git log; at least, in mine is. Let me check. Can you go ahead with Link.astro, please? Thanks!, Brayen

Ok!
I've implemented the suggested change in the interface for links (located in `OffcanvasMenu.astro`) and also fixed a bug we hadn't noticed before in `Link.astro`, related to the click event.

bjgavilanes commented:
So, I tried to push my local changes to this PR, but things happened. I put the fix from Github UI; am testing this PR at local, rn. BRB!

Thanks!,
Brayen

bjgavilanes commented:
Tested locally, works as expected. Thanks, @nsandoya!
